x-pack/plugins/rule_registry/server/utils/create_lifecycle_rule_type_factory.ts

```diff
@@ -5,50 +5,26 @@
  * 2.0.
  */
 import { Logger } from '@kbn/logging';
-import { isLeft } from 'fp-ts/lib/Either';
-import * as t from 'io-ts';
-import { Mutable } from 'utility-types';
-import v4 from 'uuid/v4';
-import { AlertInstance } from '../../../alerting/server';
 import { RuleDataClient } from '..';
 import {
   AlertInstanceContext,
   AlertInstanceState,
   AlertTypeParams,
+  AlertTypeState,
 } from '../../../alerting/common';
-import {
-  ALERT_DURATION,
-  ALERT_END,
-  ALERT_ID,
-  ALERT_START,
-  ALERT_STATUS,
-  ALERT_UUID,
-  EVENT_ACTION,
-  EVENT_KIND,
-  RULE_UUID,
-  TIMESTAMP,
-} from '../../common/technical_rule_data_field_names';
+import { AlertInstance } from '../../../alerting/server';
 import { AlertTypeWithExecutor } from '../types';
-import { ParsedTechnicalFields, parseTechnicalFields } from '../../common/parse_technical_fields';
-import { getRuleExecutorData } from './get_rule_executor_data';
+import { createLifecycleExecutor } from './create_lifecycle_executor';
 
 export type LifecycleAlertService<TAlertInstanceContext extends Record<string, unknown>> = (alert: {
   id: string;
   fields: Record<string, unknown>;
 }) => AlertInstance<AlertInstanceState, TAlertInstanceContext, string>;
 
-const trackedAlertStateRt = t.type({
-  alertId: t.string,
-  alertUuid: t.string,
-  started: t.string,
-});
-
-const wrappedStateRt = t.type({
-  wrapped: t.record(t.string, t.unknown),
-  trackedAlerts: t.record(t.string, trackedAlertStateRt),
-});
-
-type CreateLifecycleRuleTypeFactory = (options: {
+export const createLifecycleRuleTypeFactory = ({
```

Comment:
This is still used by the APM plugin but internally uses the lighter-weight executor wrapper.